std/math.d

```diff
@@ -6075,13 +6089,11 @@ public:
             return true;
         else version(ARM_Any)
         {
-            auto oldState = getControlState();
-            // If exceptions are not supported, we set the bit but read it back as zero
-            // https://sourceware.org/ml/libc-ports/2012-06/msg00091.html
-            setControlState(oldState | divByZeroException);
-            immutable result = (getControlState() & allExceptions) != 0;
-            setControlState(oldState);
-            return result;
+            // The hasExceptionTraps_impl function is basically pure,
+            // as it restores all global state
```

Comment:
This is the same reasoning used in [`core.memory.pureMalloc`](https://dlang.org/phobos/core_memory.html#.pureMalloc) etc.
>Purity is achieved by saving and restoring the value of errno, thus behaving as if it were never changed.